### ISSUE TITLE
Support numeric binary literals

### ISSUE THREAD

**Issue by ligurio** [lunarmodules/luacheck#139]
LuaJIT has extensions to the Lua parser [^1]:

> it handles ... binary (0b101010LL) literals.

Also, binary literals are supported in Luau, see [^3]

luacheck doesn't support numeric binary literals:

```lua
$ cat bin.lua 
local my_number = 0b1010
$ luacheck bin.lua 
Checking bin.lua                                  1 error

    bin.lua:2:1: expected '=' near <eof>

Total: 0 warnings / 1 error in 1 file
```

The discussion "Binary literals and number separators." [^2] may be relevant.

[^1]: https://luajit.org/ext_ffi_api.html#literals
[^2]: http://lua-users.org/lists/lua-l/2015-01/msg00196.html
[^3]: https://rfcs.luau.org/syntax-number-literals.html
